### ISSUE TITLE
plot with unsorted indices misses data points

### ISSUE THREAD

**Issue by denfromufa** [pandas-dev/pandas#11471]
Compare these 2 plots below, this is not expected behavior:

```
import pandas as pd
from numpy.random import rand
testarr1=rand(100)
testarr2=rand(100)
pd.DataFrame(testarr2,index=testarr1).plot(style='.')
pd.DataFrame(testarr2,index=testarr1).sort().plot(style='.')
```

**Comment by denfromufa** [comment]
http://stackoverflow.com/questions/12946578/is-this-a-python-pandas-dataframe-post-sort-plot-bug

**Comment by sinhrks** [comment]
Are you reporting different issue from stackoverflow? 

The difference seems to be caused by ``xlim`` adjustment logic. A workaround is:

```
ax = pd.DataFrame(testarr2,index=testarr1).plot(style='.')
ax.set_xlim(0, 1)
```

PR is welcome.

**Comment by denfromufa** [comment]
I removed SO question, it is completely unrelated after more detailed reading.
Setting `xlim` is a better workaround then `sort`.

**Comment by yoavram** [comment]
[Another example with real data](https://gist.github.com/anonymous/af438882bcf7bb6b4db9be6c720beb6c)

**Comment by jreback** [comment]
@TomAugspurger ?

**Comment by yoavram** [comment]
I'll have a go at it, but I wonder how to test it - get the data from the aces and check that xlim contains the data range ?

**Comment by TomAugspurger** [comment]
Maybe just to 2 plots, one sorted and the other not, and assert that they have the same xlim? That may work.

________________________________

**Comment by jorisvandenbossche** [comment]
Closed by https://github.com/pandas-dev/pandas/pull/16600

**Comment by yoavram** [comment]
Just want to confirm that the problem in above cited [gist](https://gist.github.com/anonymous/af438882bcf7bb6b4db9be6c720beb6c) is fixed with pandas 0.22.
